btree/btree.c

```diff
@@ -1748,7 +1754,8 @@ static void bnode_move(struct nd *src, struct nd *tgt, enum direction dir,
 
 static void bnode_capture(struct slot *slot, struct m0_be_tx *tx)
 {
-	slot->s_node->n_type->nt_capture(slot, tx);
+	struct nd* node = (struct nd *)slot->s_node;
```

Comment:
Why Do we need this change?? Also why to pass `struct node *` to `nt_capture()`, we are not using it inside those functions.

Reply:
We need this change because as a part of optimization we will keep the information of what is to be captured in the node descriptor which will be accessible to the capture routines. Also the slot information is not needed because the captures only deal with capturing of the data within the node and not specific to slots. If the record index is ever needed to help the capture then that information can be kept in some node variables to be used by the capture routines.